edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/EdgeDeploymentStatusCommand.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Agent.Core;
+    using Microsoft.Azure.Devices.Edge.Agent.Core.ConfigSources;
+    using Microsoft.Azure.Devices.Edge.Agent.Kubernetes;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class EdgeDeploymentStatusCommand : ICommand
+    {
+        readonly Option<EdgeDeploymentStatus> activeStatus;
+
+        public string Id => "EdgeDeploymentStatusCommand";
+
+        public EdgeDeploymentStatusCommand(
+            Option<EdgeDeploymentStatus> activeStatus)
+        {
+            this.activeStatus = activeStatus;
+        }
+
+        public Task ExecuteAsync(CancellationToken token)
+        {
+            // If the EdgeDeployment operator fails, report this to the plan executor
+            // as an exception.
+            this.activeStatus.ForEach(status =>
+                {
+                    if (status.State != EdgeDeploymentStatusType.Success)
+                    {
+                        throw new ConfigOperationFailureException(status.Message);
```

Comment:
Besides failure message, what can be the inner exception or callstack? It would be useful for troubleshooting.

Reply:
I chose not to report the full call stack, since this will eventually put into the twin, which has size limits.